content/en/about-types.md

```diff
@@ -0,0 +1,261 @@
+---
+title: Types
+description: The JSDoc type system.
+---
+
+JSDoc allows you to specify type expressions identifying the type of value that a symbol may
+contain, a parameter type, or the type of a value returned by a function. These type expressions
+are specified with the [`@type` tag][type-tag], the [`@param` tag][param-tag], and many others.
+
+[type-tag]: tags-type.html
+[param-tag]: tags-param.html.
+
+## Overview
+
+A type expression can include the JSDoc namepath to a symbol (for example, `myNamespace.MyClass`); a
+built-in JavaScript type (for example, `string`); or a combination of these. You can use any
+[Google Closure Compiler type expression][closure], as well as several other formats that are
+specific to JSDoc.
+
+If JSDoc determines that a type expression is invalid, it will display an error and stop running.
+You can turn this error into a warning by running JSDoc with the `--lenient` option.
+
+**Note**: Full support for Google Closure Compiler-style type expressions is available
+in JSDoc 3.2 and later. Earlier versions of JSDoc included partial support for Closure Compiler type
+expressions.
+
+Each type is specified by providing a type expression, using one of the formats described below.
+Where appropriate, JSDoc will automatically create links to the documentation for other symbols. For
+example, `@type {MyClass}` will link to the MyClass documentation if that symbol has been
+documented.
+
+<table id="jsdoc-types" name="jsdoc-types">
+<tr>
+ <th>Type name</th>
+ <th>Syntax examples</th>
+ <th>Description</th>
+</tr>
+
+<tr>
+ <td>Symbol name (name expression)</td>
+ <td>
+{% example %}
+<pre class="prettyprint"><code>{boolean}
+{myNamespace.MyClass}
+</code></pre>
+{% endexample %}
+ </td>
+ <td>
+  <p>
+  Specifies the name of a symbol. If you have documented the symbol, JSDoc creates a link to the
+  documentation for that symbol.
+  </p>
+ </td>
+</tr>
+
+<tr>
+ <td>
+  Multiple types (type union)
+ </td>
+ <td>
+{% example "This can be a number or a boolean." %}
+<pre class="prettyprint"><code>{(number|boolean)}
+</code></pre>
+{% endexample %}
+ </td>
+ <td>
+  <p>
+  This means a value can have one of several types, with the entire list of types enclosed in
+  parentheses and separated by <code>|</code>. The parentheses are required for Closure Compiler,
```

Comment:
Is this accurate? I thought Closure Compiler supported type unions without parentheses, even though that behavior isn't documented.

Reply:
[This doc](https://github.com/google/closure-compiler/wiki/Annotating-JavaScript-for-the-Closure-Compiler#type-union) says that they're required, but we then ignore that rule most other places. So you're right, I'll make it clear that they're optional by saying Closure "suggests" them.